modules/dnn/src/onnx/onnx_importer2.cpp

```diff
@@ -0,0 +1,1714 @@
+    //dispatch["Pad"] = &OnnxImporter2::parsePad;
+    dispatch["PRelu"] = &OnnxImporter2::parsePRelu;
+    dispatch["ReduceMax"] = dispatch["ReduceMin"] = dispatch["ReduceMean"] = dispatch["ReduceSum"] = dispatch["ReduceMax"] =
+    dispatch["ReduceMin"] = dispatch["ReduceSumSquare"] = dispatch["ReduceProd"] = dispatch["ReduceL1"] =
+    dispatch["ReduceL2"] = dispatch["ReduceLogSum"] = dispatch["ReduceLogSumExp"] = &OnnxImporter2::parseReduce;
+    dispatch["Relu"] = &OnnxImporter2::parseRelu;
+    dispatch["Reshape"] = &OnnxImporter2::parseReshape;
+    dispatch["Resize"] = &OnnxImporter2::parseResize;
+    dispatch["Shape"] = &OnnxImporter2::parseShape;
+    dispatch["Slice"] = &OnnxImporter2::parseSlice;
+    dispatch["Softmax"] = dispatch["SoftMax"] = dispatch["LogSoftmax"] = &OnnxImporter2::parseSoftMax;
+    dispatch["SpaceToDepth"] = dispatch["DepthToSpace"] = &OnnxImporter2::parseDepthToSpace;
+    dispatch["Split"] = &OnnxImporter2::parseSplit;
+    dispatch["Squeeze"] = &OnnxImporter2::parseSqueeze;
+    dispatch["Transpose"] = &OnnxImporter2::parseTranspose;
+    //dispatch["Upsample"] = &OnnxImporter2::parseUpsample;
+    dispatch["Unsqueeze"] = &OnnxImporter2::parseUnsqueeze;
+
+    vector<string> simpleLayers{
+        "Abs", "Acos", "Acosh", "Asin", "Asinh", "Atan", "Atanh",
+        "Ceil", "Celu", "Cos", "Cosh", "Elu", "Erf",
+        "Exp", "Floor", "HardSigmoid", "HardSwish", "Identity",
+        "Log", "Neg", "Round", "Reciprocal", "Selu",
+        "Sign", "Sigmoid", "Sin", "Sinh", "Softplus",
+        "Softsign", "Shrink", "Sqrt", "Tan", "Tanh", "ThresholdedRelu"};
+    for (const auto& name : simpleLayers)
+        dispatch[name] = &OnnxImporter2::parseElemwiseUnary;
+
+    // ai.onnx: opset 10+
+    dispatch["DequantizeLinear"] = &OnnxImporter2::parseDequantizeLinear;
+    dispatch["QLinearConv"] = &OnnxImporter2::parseQLinearConv;
+    dispatch["QLinearMatMul"] = &OnnxImporter2::parseQLinearMatMul;
+    dispatch["QuantizeLinear"] = &OnnxImporter2::parseQuantizeLinear;
+    alldispatch[defaultOnnxDomain] = dispatch;
+    alldispatch[""] = dispatch;
+
+    msdispatch["QLinearAdd"] = &OnnxImporter2::parseQLinearElemwiseBinary;
+    msdispatch["QLinearAveragePool"] = &OnnxImporter2::parsePooling;
+    msdispatch["QLinearGlobalAveragePool"] = &OnnxImporter2::parseQLinearGlobalAveragePool;
+    msdispatch["QLinearConcat"] = &OnnxImporter2::parseQLinearConcat;
+    msdispatch["QLinearLeakyRelu"] = &OnnxImporter2::parseQLinearLeakyRelu;
+    msdispatch["QLinearSigmoid"] = &OnnxImporter2::parseQLinearSigmoid;
+    alldispatch["com.microsoft"] = msdispatch;
+}
+
+static inline void replaceLayerParam(LayerParams& layerParams, const String& oldKey, const String& newKey)
+{
+    if (layerParams.has(oldKey)) {
+        layerParams.set(newKey, layerParams.get(oldKey));
+        layerParams.erase(oldKey);
+    }
+}
+
+/*LayerParams OnnxImporter2::getLayerParams(const opencv_onnx::NodeProto& node_proto)
```

Comment:
You are planning to remove it, right?
Anakin.jpg

Reply:
right. I removed this chunk of code from ONNX importer v2. will take the necessary code from the old ONNX importer v1 when needed